daily_milestone.go

```diff
@@ -1,29 +1,100 @@
 package main
 
-import "flag"
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+)
 
-func get_project_id(base_url string, token string, projectname string, namespace string) string {
-	headers := map[string]string{"PrivateToken": token}
-	url := "https://" + base_url + "/projects"
+var logger *log.Logger
+
+//LoggerSetup Initialization
+func LoggerSetup(info io.Writer) {
+	logger = log.New(info, "INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
+}
+
+func getProjectID(BaseURL string, Token string, Projectname string, Namespace string) string {
+	urls := "https://" + BaseURL + "/projects"
 	page := 1
+	headers := url.Values{}
```

Comment:
Instead of using headers and overwriting them with url.Values, we can actually create a new http post request with something like this: 
```
    client := &http.Client{}
    r, _ := http.NewRequest("POST", url, bytes.NewBufferString(data.Encode()))
    r.Header.Add("Token", "")
```

Reply:
Just looked it up: Because we don't post/push anything to the server yet, we can simply use a Get request.
    r, _ := http.NewRequest("GET", url)

Reply:
Another example would be:
```

u, err := url.Parse(ourUrl) // parses the url so we can more easily work with it
if err != nil {
    log.Fatal(err)
}
q := u.Query() // returns the current query could be something like ?something=1&something=else
q.Add("page", page) // we add our data/string to the query this automatically handles choosing between ? and &
// q.Set would replace a value with a specific key more docs: https://golang.org/pkg/net/url/#ParseQuery
u.RawQuery = q.Encode() // this will encode the string to conform to url encoding
resp, err := http.Get(u) // we will use this url to send a get request.
```

Reply:
Oh thanks a lot this was bugging me I was unable to get the understanding of newRequest tried using it first but documentation was unclear. Thanks a lot.